src/main.ts

```diff
@@ -41,152 +45,140 @@ const getPGConfig = (): E.Either<Error, ClientConfig> =>
-  messagingClient: KafkaProducerCompact<Student>
-): TE.TaskEither<Error, void> =>
-  pipe(
-    process.stdin.resume(),
-    void process.on("SIGINT", () => {
-      void cleanupAndExit({
-        pgClient: dbClient,
-        kafkaClient: messagingClient,
-      })();
-    })
-  );
+const waitForExit =
+  (): RTE.ReaderTaskEither<Deps, Error, void> =>
+  ({ pgClient, messagingClient }: Deps) =>
+    pipe(
+      process.stdin.resume(),
+      void process.on("SIGINT", () => {
+        cleanupAndExit({
+          pgClient,
+          messagingClient,
+        })().catch(error(`Error during the exit`));
+      })
+    );
 
-const cleanupAndExit = (clients: {
-  pgClient: PGClient;
-  kafkaClient: KafkaProducerCompact<Student>;
-}): TE.TaskEither<Error, void> =>
-  pipe(
-    disconnectPGLogicalClient(clients.pgClient),
-    TE.chain(() => query(clients.pgClient, QUERIES.DROP_PUBLICATION)),
-    TE.chain(() =>
-      query(clients.pgClient, QUERIES.DROP_LOGICAL_REPLICATION_SLOT)
-    ),
-    TE.chain(() => disconnectPGClient(clients.pgClient)),
-    TE.chain(() =>
-      pipe(
-        clients.kafkaClient,
-        TE.fromIO,
-        TE.bindTo("client"),
-        TE.chainFirst(({ client }) => disconnect(client.producer))
-      )
-    ),
-    TE.map(() => {
-      pipe(
-        defaultLog.taskEither.info(
-          "Disconnected from Database and Message Bus"
-        ),
-        process.exit(0)
-      );
-    }),
-    TE.mapLeft((error) => {
-      defaultLog.taskEither.error(`Error during the exit - ${error}`);
-      process.exit(1);
-    })
-  );
+const disconnectKafkaProducer =
+  () =>
+  ({ messagingClient }: QueueDeps) =>
+    pipe(
+      messagingClient,
+      TE.fromIO,
+      TE.chain((client) => disconnect(client.producer))
+    );
+
+const cleanupAndExit = pipe(
+  disconnectPGLogicalClient(),
+  RTE.chainFirst(() => query(QUERIES.DROP_PUBLICATION)),
+  RTE.chainFirst(() => query(QUERIES.DROP_LOGICAL_REPLICATION_SLOT)),
+  RTE.chainFirst(disconnectPGClient),
+  RTE.chainFirstW(disconnectKafkaProducer),
+  RTE.chainFirst(() => {
+    info("Disconnected from Database and Message Bus");
+    process.exit(0);
+  }),
+  RTE.mapLeft((e) => {
+    error(`Error during the exit - ${e}`);
+    process.exit(1);
+  })
+);
 
-const exitFromProcess = (): TE.TaskEither<Error, void | object> =>
-  pipe(defaultLog.taskEither.error("Application failed"), process.exit(1));
+// const exitFromProcess = (): TE.TaskEither<Error, void | object> =>
+//   pipe(defaultLog.taskEither.error("Application failed"), process.exit(1));
 
-const main = () =>
+const main = pipe(
+  RTE.Do,
+  RTE.chainFirstW(
+    logRTE.info("Trying to connect to the event hub instance...")
+  ),
+  RTE.bind("messagingClient", () => RTE.fromEither(getEventHubProducer())),
+  RTE.chainFirstW(logRTE.info("Connected to the event hub instance")),
+  RTE.chainFirstW(logRTE.info("Creating PostgreSQl client...")),
+  RTE.bind("pgClient", createPGClient),
+  RTE.map(
```

Comment:
I'm trying to run the code but it seems to stuck at line 150. After the creation of the PGClient the run is interrupted without errors.
I've checked that createPGClient ends succesfully and it happens.